### PR TITLE
feat: complete task-04

### PR DESCRIPTION
**Pull request by aliskar**
**Evaluation Criteria**

2. All tasks are implemented partially, or two tasks are not implemented at all.  
3. One task is not implemented at all, or two tasks have major issues.
4. One of the tasks does not have a major part of its implementation.  
5. All tasks are implemented to a full extend.  
5*. Typescript decorator has been used to implement Step 4.

### CODE REVIEW

File: 04_design_patterns/.prettierrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "trailingComma": "es5",
+    "tabWidth": 2,
+    "semi": false,
+    "singleQuote": true,
+    "printWidth": 120
+  }
```
Comment (byshyk) [inline 04_design_patterns/.prettierrc:7]: add new line, fix indent

File: 04_design_patterns/jest.config.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+    preset: 'ts-jest',
+    testEnvironment: 'node',
+  };
```
Comment (byshyk) [inline 04_design_patterns/jest.config.ts:4]: add new line, fix indent

File: 04_design_patterns/src/Shipments/Package.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+export class Package extends Shipment {
+  constructor(state: ShipmentState) {
+    super(state)
+  }
+
+  public getCost() {
+    return this.shipmentStrategy.getCost(this.getWeight(), ShipmentType.PACKAGE)
+  }
+}
```
Comment (byshyk) [inline 04_design_patterns/src/Shipments/Package.ts:12]: add new line

File: 04_design_patterns/.prettierrc (hunk 1 above)
Comment (aliskar) [reply 04_design_patterns/.prettierrc]: done

**Pull request merged**